vllm/v1/attention/backends/flash_attn.py

```diff
@@ -477,7 +545,7 @@ def forward(
             block_table = attn_metadata.block_table
             scheduler_metadata = attn_metadata.scheduler_metadata
 
-            descale_shape = (cu_seqlens_q.shape[0] - 1, key.shape[1])
+            descale_shape = (cu_seqlens_q.shape[0] - 1, self.num_kv_heads)
```

Comment:
I was wondering why this change is required.

Reply:
it might be worth adding this comment around why the `key is not None and value is not None` is needed above because this clarified a lot for me